client/js/render.js

```diff
@@ -90,6 +99,54 @@ function buildChatMessage(data) {
 	return msg;
 }
 
+function renderPreview(preview, msg) {
```

Comment:
Can we keep all this preview code in `msg_preview`, otherwise we will end up having this render file huge again.